pace-util/pace/util/decomposition.py

```diff
@@ -0,0 +1,141 @@
+from __future__ import annotations
+
+import os
+from typing import TYPE_CHECKING, Tuple
+
+from gt4py import config as gt_config
+
+from pace.util import TilePartitioner
+from pace.util.partitioner import CubedSpherePartitioner
+
+
+if TYPE_CHECKING:
+    from pace.dsl.stencil_config import CompilationConfig
```

Comment:
yeah:
decomposition is a helper file to clarify functionality used in stencil_config.
On one side it has `block_waiting_for_compilation` which takes that a `compilation_config`
On the other side it is used in the `__init__` from the `CompilationConfig` calls `get_decomposition_info_from_comm`.
Since there is no actual circular dependency here there are 3 options:

1. Do not put a typehint in there
2. Reoder the functions (put some in stencil_config, put some in a separate file)
3. Do this

I felt like option 1 is not really an option and the functionality here is tight enough together that I'd rather do it this way. We are doing this also in other places, see [here](https://github.com/ai2cm/pace/blob/1cf72fcb81f83710f57fe415b4dcbace4862d722/pace-util/pace/util/halo_updater.py#L21) for example